test/diagonalGate.jl

```diff
@@ -0,0 +1,220 @@
+using Snowflake
+using Test
+using StaticArrays
+
+@testset "Diagonal Gate: phase_shift" begin
+
```

Comment:
I would like to refactor the tests a bit.

Testsets allow grouping and naming tests.

Can we try the following structure?

* "Phase shift gate"
  * "Apply gate is equivalent to apply phase shift operator"
  * "Constructor"
  * "Apply gate applied on the last qubit"
  * "Apply gate applied on a non-last qubit"
  * "Adjoint"
  * "Get inverse"
  * "Legacy" (Useful to know which functionality won't be there in the final version.)
  
 Adapt as it is necessary, but then it is very clear what we are testing, and then we don't need comments for any test groups. Naturally, this also applies to the rest of the tests for "new" gates.